sdks/python/apache_beam/pipeline.py

```diff
@@ -772,6 +772,14 @@ def _infer_result_type(
       input_element_type = (
           input_element_types_tuple[0] if len(input_element_types_tuple) == 1
           else typehints.Union[input_element_types_tuple])
+      input_batch_type = inputs[0].batch_type
+
+      # All defined batch_types should be equivalent
+      # TODO: Nicer error
```

Comment:
There can't actually be multiple inputs to a ParDo. We should explicitly test the Flatten case